app/Http/Controllers/Admin/BeneficiaryController.php

```diff
@@ -193,118 +195,83 @@ public function update_call_champion_report($due_list){
 	public function list_all_beneficiaries(){
 		
 	}
-	
-	public function importExcel(){
-    	$beneficiary= new Beneficiary;
-    	$mess=array();
-    	$destinationPath = 'external/uploads'; // upload path
-    	$extension = Input::file('txtExcel')->getClientOriginalExtension(); // getting image extension
-    	$fileName = rand(11111,99999).'.'.$extension; // renameing image
-    	Input::file('txtExcel')->move($destinationPath, $fileName); // uploading file to given path
-    	
-  	// Applying validation rules.
-    $filedworkerid="";
-    if($_POST['fieldwrokerid']!="")
-    	$filedworkerid=$this->decode($_POST['fieldwrokerid']);
-  	$rules = array(
-  			'name' => 'required|min:3|max:20|Regex:/^[A-Za-z]+[A-Za-z0-9.\' ]*$/',
-  			'husband_name'=>'required|min:3|max:20',
-  			'phone_number'=>'required|numeric|digits_between:10,20',
-  			'village'=>'required|min:3|max:20',
-  			'language'=>'required',
-  			'pincode'=>'required|numeric|digits_between:4,6',
-  			'due_date'=>'required_without:delivery_date|date_format:d/m/Y',
-  			'delivery_date'=>'required_without:due_date|date_format:d/m/Y'
-  			
-  	);
-  		$arr = Excel::load($destinationPath.'/'.$fileName)->get();
-    	$i=0;
-    	if(!empty($arr)){
-    		if(isset($arr[0]->name) && isset($arr[0]->language) && isset($arr[0]->number_pregnancies) && isset($arr[0]->husband_name) && isset($arr[0]->phone_number) && isset($arr[0]->alternate_phone_no)  && isset($arr[0]->village) && isset($arr[0]->pincode)   && isset($arr[0]->taluka) && isset($arr[0]->awc_name) && isset($arr[0]->awc_number) && isset($arr[0]->due_date) && isset($arr[0]->delivery_date)){
-    		
-    		}else{
-    			Session::flash('message', '<div class="alert alert-error" style="clear:both;">
-              <button data-dismiss="alert" class="close" type="button">×</button>'.trans("routes.excelvalid").'</div>');
-    			return Redirect::to('/beneficiary/');
-    		}
-    		foreach($arr as $key=>$val){
-    			$datetime = date("Y-m-d H:i:s");
-    			$i++;
-    			$due_date=str_replace("-", "/", $val->due_date);
-    			$delivery_date=str_replace("-", "/", $val->delivery_date);
-    			$userdata = array(
-    					'name' => $val->name,
-    					'husband_name'=>$val->husband_name,
-    					'phone_number'=>$val->phone_number,
-    					'village'=>$val->village,
-    					'language'=>$val->language,
-    					'pincode'=>$val->pincode,
-    					'due_date'=>date('d/m/Y',strtotime($due_date)),
-    					'delivery_date'=>date('d/m/Y',strtotime($delivery_date))
+	public function downloadExcel()
+	{
+		return response()->download(public_path('download\sample.xlsx'));
+	}
+	public function importExcel()
+	{
+		Session::flash('count',0);		
```

Comment:
@Shubham2801 

can you rename this to something more meaningful then? Like count_excelupload_errors
count can be confusing for others who encounter this variable at any other place.
please change, test on your local thoroughly and recommit.
will merge your changes.